src/Framework/Framework/Resources/Scripts/utils/evaluator.ts

```diff
@@ -1,18 +1,51 @@
 import { isObservableArray } from "./knockout";
 import { logError } from "./logging";
 
-export function evaluateOnViewModel(context: any, expression: string): any {
-    // todo: reimplement
-    let result;
-    if (context && context.$data) {
-        result = new Function("$context", "with($context) { with ($data) { return " + expression + "; } }")(context);
-    } else {
-        result = new Function("$context", "var $data=$context; with($context) { return " + expression + "; }")(context);
+const shouldBeConvertedFromDataContext = (currentLevel: any, remainingParts: string[]): boolean => {
+    if (currentLevel["$data"] == undefined) {
+        return false;
     }
-    if (result && result.$data) {
-        result = result.$data;
+
+    return remainingParts.filter(i => i.startsWith("$")).length === 0;
+};
+
+
+
+/**
+ * Traverses provided context according to given path.
+ * @example / - returns context
+ * @example "" or null - returns context
+ * @example /exampleProp/A - returns prop A located withing property exampleProp located at provided context
+ * @example /exampleProp/B/1 - returns second item from collection B located within property exampleProp located at provided context
+ * @returns found property as unwrapped object
+ */
+export function traverseContext(context: any, path: string): any 
+{
```

Comment:
If I understand this function correctly, path = `/` will return the current view model and not the root. Am I wrong?

Reply:
`/` returns provided context.
Based on validation target this may or may not be root.
Root VM is used for application of validation errors from server.

DataContext of origin element (sender) is used for client side validation.

Reply:
This is already resolved. `/` always means root viewmodel. Same as `validate-root`. Moreover, `validate-this` is removed